File: gpp-api/src/main/java/com/develop/gpp/domain/entity/ItemPedidosaidaModel.java
package com.develop.gpp.domain.entity;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;

@Entity
public class ItemPedidosaidaModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idItemPedidoSaida;
    
    private int quantidade;
    
    @OneToOne
    private PecaModel peca;
    
    private int separado;
    
    @OneToOne
    private FuncionarioModel separador;
    
    private Date dataSeparacao;
    
    @OneToOne
    private AstecaMotivoModel motivoTrocaPeca;
    
    private boolean pendenciaItem;
    
    private String endereco;
    
    private int saldo_disponivel;
    
    @OneToOne
    private PecasEstoqueModel pecaEstoque;
    
    // Add getters and setters
}

File: angular-gpp/src/app/models/ItemPedidosaidaModel.ts
import { PecaModel } from "./PecaModel";

export class ItemPedidosaidaModel {
  idItemPedidoSaida: number = 0;
  quantidade: number = 0;
  peca: PecaModel = new PecaModel;
  separado: number = 0;
  separador: FuncionarioModel = new FuncionarioModel;
  dataSeparacao!: Date;
  motivoTrocaPeca: MotivoModel = new MotivoModel;
  pendenciaItem: boolean = false;
  endereco: String;
  saldo_disponivel: int;
  pecaEstoque: PecasEstoqueModel = new PecasEstoqueModel;

}
